File: src/main/java/ast/ASTRequester.java
package ast;

import com.google.inject.Guice;
import com.google.inject.Inject;
import com.google.inject.Injector;
import org.eclipse.jdt.core.dom.CompilationUnit;
import org.eclipse.jdt.core.dom.FileASTRequestor;
import utils.inject.DependencyManager;

public class ASTRequester extends FileASTRequestor {

    @Inject
    private final ASTVisitor astVisitor;

    public ASTRequester() {
        Injector injector = Guice.createInjector(new DependencyManager());
        this.astVisitor = injector.getInstance(ASTVisitor.class);
    }

    @Override
    public void acceptAST(String sourceFilePath, CompilationUnit ast) {
        System.out.println("Scanning file " + sourceFilePath);
        ast.accept(astVisitor);
    }

}

File: src/main/java/org/vladg/jafax/ast/ASTRequester.kt
package org.vladg.jafax.ast

import com.google.inject.Inject
import org.eclipse.jdt.core.dom.CompilationUnit
import org.eclipse.jdt.core.dom.FileASTRequestor
import org.vladg.jafax.utils.extensions.convertFilePathToUniversalPath
import org.vladg.jafax.utils.extensions.logger

class ASTRequester : FileASTRequestor() {

    private val logger = logger()

    @Inject
    private lateinit var astVisitor: ASTVisitor

    override fun acceptAST(sourceFilePath: String, ast: CompilationUnit) {
        val universalPath = convertFilePathToUniversalPath(sourceFilePath)
        logger.info("Scanning file: $universalPath")
        astVisitor.currentFileName = universalPath
        try {
            ast.accept(astVisitor)
        } catch (ex: Exception) {
            logger.error("There was an error while ast parsing, will skip that entry ...", ex)
        }
    }

}